apps/client/src/routes/dashboards/dashboards-index/dashboards-index-page.tsx

```diff
@@ -0,0 +1,187 @@
+
+import { useDashboardsTable } from './hooks/use-dashboards-table';
+import { useDeleteModalVisibility } from './hooks/use-delete-modal-visibility';
+import { usePartialUpdateDashboardMutation } from './hooks/use-partial-update-dashboard-mutation';
+import { DashboardsTableHeader } from './components/dashboards-table-header';
+import { DeleteDashboardModal } from './components/delete-dashboard-modal';
+
+export function DashboardsIndexPage() {
+  const intl = useIntl();
+  const [isDeleteModalVisible, setIsDeleteModalVisible] =
+    useDeleteModalVisibility();
+  const updateDashboardMutation = usePartialUpdateDashboardMutation();
+  const { selectedItems, tableProps } = useDashboardsTable();
+
+  return (
+    <>
+      <Table
+        {...tableProps}
+        ariaLabels={{
+          itemSelectionLabel: (_selectionState, row) =>
+            intl.formatMessage(
+              {
+                defaultMessage: 'Select dashboard {name}',
+                description: 'dashboards table selection group label',
+              },
+              { name: row.name },
+            ),
+        }}
+        submitEdit={async (item, column, newValue) => {
+          invariant(column.id, 'Expected column to be defined');
+
+          await updateDashboardMutation.mutateAsync({
+            id: item.id,
+            [column.id]: newValue,
+          });
+        }}
+        header={
+          <DashboardsTableHeader
+            isDeleteDisabled={selectedItems.length === 0}
+            onClickDelete={() => setIsDeleteModalVisible(true)}
+          />
+        }
+        columnDefinitions={[
+          {
+            id: 'dashboard-id',
+            header: intl.formatMessage({
+              defaultMessage: 'ID',
+              description: 'dashboards table ID column header',
+            }),
+            cell: (dashboard) => dashboard.id,
+            sortingField: 'dashboard-id',
+          },
+          {
+            id: 'name',
+            header: intl.formatMessage({
+              defaultMessage: 'Name',
+              description: 'dashboards table name column header',
+            }),
+            cell: (dashboard) => (
+              <Link to={`/dashboards/${dashboard.id}`}>{dashboard.name}</Link>
+            ),
+            sortingField: 'name',
+            editConfig: {
+              ariaLabel: intl.formatMessage({
+                defaultMessage: 'Name',
+                description: 'dashboard table name edit cell aria label',
+              }),
+              editIconAriaLabel: intl.formatMessage({
+                defaultMessage: 'Name Editable',
+                description:
+                  'dashboard table name edit cell edit icon aria label',
+              }),
+              errorIconAriaLabel: intl.formatMessage({
+                defaultMessage: 'Name Error',
+                description:
+                  'dashboard table name edit cell error icon aria label',
+              }),
+              editingCell: (
```

Comment:
is validation handled? if not, `<Table>` has this `editConfig.validation` for displaying an error message

Reply:
Good catch, need to look into this more.